### PR TITLE
try to find spec file in some subdirectories too

### PR DESCRIPTION
**Pull request by xsuchy**
Some projects hesitate to put spec file in top levele directories and use "packaging" or "dist". Try to look there too.

### CODE REVIEW

**Comment by xsuchy** [comment]
ping?

File: src/tito/common.py [hunk 1]
```diff
@@ -229,20 +229,31 @@ def create_builder(package_name, build_tag,
...
         if f.endswith(suffix):
             if file_name is not None:
                 error_out("At least two %s files in directory: %s and %s" % (suffix, file_name, f))
             file_name = f
-            debug("Using file: %s" % f)
+            debug("Using file: %s" % os.path.join(in_dir, f))
     if file_name is None:
-        error_out("Unable to locate a %s file in %s" % (suffix, in_dir))
+        if top:
+            sub_dir_list = ["distribution", "dist", "packaging"]
```
Comment (jumanjiman) [inline src/tito/common.py]: i also see projects put the spec file in `contrib` and `spec` subdirs

Comment (jumanjiman) [inline src/tito/common.py]: perhaps there should be a tito property to specify the subdir in which spec file lives?

Comment (dgoodwin) [inline src/tito/common.py]: This alone is probably a pretty good sign we shouldn't try to guess the location, too many possibilities.

For a single project repo, dropping into tito.props seems like a good idea. That won't work so hot for multi-project repos though. Perhaps we define in tito.props, and assume to look in that sub-dir relative to the package dir.

**Comment by dgoodwin** [comment]
Apologies for the delay here but this really needs unit tests, otherwise I'm going to end up breaking it or merging something that does and then not realize. 

Could you add some unit tests to verify it's working for both singleproject and multi-project tito repos? Should be good examples to start with in test/functional/singleproject_tests.py and multiproject_tests.py.

File: src/tito/common.py (hunk 1 above)
Comment (dgoodwin) [inline src/tito/common.py]: Also would searching for a spec file be reasonable?

Comment (jumanjiman) [inline src/tito/common.py]: let's see we search and find a `.spec` file.
we then need to figure out:

* is it really an rpm spec file?
* where does the spec file live in relation to the top of its dir tree, especially wrt multi-project repo

Comment (xsuchy) [inline src/tito/common.py]: > Also would searching for a spec file be reasonable?

Rather not. That could be just example. Test file... whatever. However I agree with configuration variable in tito.props.

**Comment by dgoodwin** [comment]
Going to close for now, please reopen when ready if this is to be pursued.

**Pull request closed without merging**